oecophylla/cli/launch.py

```diff
@@ -86,67 +122,97 @@ def _create_dir(_path):
     snakefile = "%s/../../Snakefile" % os.path.abspath(os.path.dirname(__file__))
 
-    # INPUT DIR
-    for inp_file in glob.glob('%s/*' % input_dir):
-
-        file_format = sniff(inp_file)[0]
-
-        if (file_format != 'fasta') and (file_format != 'fastq'):
-            raise TypeError('Input files need to be in FASTA or FASTQ format.')
-
-    # OUTPUT
-    # create output directory, if does not exist
-    _create_dir(output_dir)
-
-    # SAMPLE SHEET
-    # If a manifest is not specified, the files containing forward and reverse reads
-    # will be automatically identified using regex.
-    if sample_sheet:
-        _sheet = read_sample_sheet(sample_sheet)
-        sample_dict = extract_samples_from_sample_sheet(_sheet, input_dir)
+    # Check to see if running with test data. If so, fill in defaults
+    # for relevant empty parameters.
+    if test:
+        input_dir, sample_sheet, params, envs, output_dir = _setup_test()
+    elif not output_dir:
+        raise IOError("Must provide output directory to run.")
+
+    # Check to see if config.yaml exists in output dir. If it does, warn
+    # and continue with execution
+    if os.path.exists(os.path.join(output_dir,'config.yaml')):
+        config_fp = '%s/%s' % (output_dir, 'config.yaml')
+    # Otherwise, make a config file and continue with execution
+    elif (os.path.exists(params) and
+          os.path.exists(envs) and
+          os.path.exists(input_dir)):
+        # INPUT DIR
+        for inp_file in glob.glob('%s/*' % input_dir):
+
+            file_format = sniff(inp_file)[0]
+
+            if (file_format != 'fasta') and (file_format != 'fastq'):
+                raise TypeError('Input files need to be in FASTA'
+                                ' or FASTQ format.')
+
+        # OUTPUT
+        # create output directory, if does not exist
+        _create_dir(output_dir)
+
+        # SAMPLE SHEET
+        # If a manifest is not specified, the files containing forward and
+        # reverse reads will be automatically identified using regex.
+        # Input dir will be converted to abspath.
+
+        input_dir = os.path.abspath(input_dir)
+
+        if sample_sheet:
+            _sheet = read_sample_sheet(sample_sheet)
+            sample_dict = extract_samples_from_sample_sheet(_sheet, input_dir)
+        else:
+            sample_dict = extract_sample_paths(input_dir)
+
+        # PARAMS
+        with open(params, 'r') as f:
+            params_dict = yaml.load(f)
+
+        # ENVS
+        with open(envs, 'r') as f:
+            envs_dict = yaml.load(f)
+
+        # CONFIG
+        # merge PARAMS, SAMPLE_DICT, ENVS
+        config_dict = {}
+        config_dict['samples'] = sample_dict
+        config_dict['params'] = params_dict
+        config_dict['envs'] = envs_dict
+        config_dict['tmp_dir_root'] = local_scratch
+
+        config_yaml = yaml.dump(config_dict, default_flow_style=False)
+
+
+        config_fp = '%s/%s' % (output_dir, 'config.yaml')
```

Comment:
Though does not matter in Linux, shall we uniformize the usages of `os.path.join` and '%s/%s'?

Reply:
👍